Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -1,11 +1,54 @@
+import { useEffect, useState } from "react";
+
 import BatteryInput from "~/components/batteryInput";
+import CalculateButton from "~/components/calculateButton";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
+  const [speedInput, setSpeedInput] = useState("");
+  const [batteryInput, setBatteryInput] = useState("");
+  const [weatherInput, setWeatherInput] = useState("");
+  const [calculatedRange, setCalculatedRange] = useState();
+  const [isClicked, setIsClicked] = useState(false);
+  // console.log(speedInput);
+  // console.log(batteryInput);
+  // console.log(weatherInput);
+
+  const calculateRange = (s, b, w) => {
+    console.log(s);
+    console.log(b);
+    if (s == "") {
+      return ["Speed is required", "text-red-500"];
```

Comment:
Returning as array is not typesafe, try something else 

One way to fix this is to instead only pass the error string, but conditionally render the "red text" class using state/derived state